### PR TITLE
Improves DKG and adds new functionalities

### PR DESCRIPTION
**Pull request by nkcr**
Most of this work is based on @Mahsa-Bastankhah's work. 

- Improves the overall workflow and make the code more efficient
- Adds the resharing functionality
- Adds verifiable encryption/decryption

~This work contains research-specific work (F3B) and should not be merged as-is.~

Built from https://github.com/dedis/dela/pull/230 and https://github.com/dedis/dela/pull/231.

### CODE REVIEW

File: test/F3B_test.go [hunk 1]
```diff
@@ -0,0 +1,351 @@
...
+	"go.dedis.ch/kyber/v3/xof/keccak"
+)
+
+func init() {
+	rand.Seed(0)
+}
+
+func Test_F3B(t *testing.T) {
+	batchSizes := []int{1}
+	// numDKGs := []int{3, 10, 20, 30, 50, 70, 100}
```
Comment (pierluca) [inline test/F3B_test.go:45]: commented out code ?

File: test/F3B_test.go [hunk 2]
```diff
@@ -0,0 +1,351 @@
+package integration
```
Comment (pierluca) [inline test/F3B_test.go:1]: this file is not F3B related, we should probably rename it.

File: test/F3B_test.go [hunk 3]
```diff
@@ -0,0 +1,351 @@
...
+	"go.dedis.ch/kyber/v3"
+	"go.dedis.ch/kyber/v3/suites"
+	"go.dedis.ch/kyber/v3/xof/keccak"
+)
+
+func init() {
+	rand.Seed(0)
+}
+
+func Test_F3B(t *testing.T) {
```
Comment (pierluca) [inline test/F3B_test.go:43]: Remove F3B references and rename appropriately

**Review by pierluca — CHANGES_REQUESTED** [review]

File: test/F3B_test.go (hunk 3 above)
Comment (nkcr) [reply test/F3B_test.go]: The F3B test has been removed. It didn't make sense to keep it.

**Comment by nkcr** [comment]
Added unit tests to maintain our coverage. You can check the last few commits to see the changes.

**Pull request merged**